### PR TITLE
More OCR spec validation

### PR DESCRIPTION
**Pull request by connorwstein**
- Support sane defaults for intervals/timeouts 
- Enforce required params for bootstrap/non-bootstrap nodes
- Enforce that non-bootstrap params are not explicity set for bootstrap nodes (and vice versa). This includes the same checks that were added in https://github.com/smartcontractkit/chainlink/pull/3597 (albeit removing the timeout ones as those now default) but just doing a similar thing for non-bootstrap. 
- Enforce the same range requirements the libocr does so we can fail fast before creating a bad job. TODO expose the validate local config function from libocr to keep that defined in one place.
- Enforce valid multiaddrs

### CODE REVIEW

File: core/services/validators.go [hunk 1]
```diff
@@ -392,36 +408,119 @@ func ValidatedOracleSpec(tomlString string) (spec offchainreporting.OracleSpec,
...
+	if time.Duration(spec.ContractConfigTrackerPollInterval) < 15*time.Second || time.Duration(spec.ContractConfigTrackerPollInterval) > 120*time.Second {
+		return spec, errors.Errorf("require 15s <= contract config tracker poll interval <= 120s ")
 	}
-	return
+	if time.Duration(spec.ContractConfigTrackerSubscribeInterval) < 2*time.Minute || time.Duration(spec.ContractConfigTrackerSubscribeInterval) > 5*time.Minute {
+		return spec, errors.Errorf("require 2m <= contract config subscribe interval <= 5m ")
+	}
+	if spec.ContractConfigConfirmations < 2 || spec.ContractConfigConfirmations > 10 {
+		return spec, errors.Errorf("require 2 <= contract config confirmations <= 10 ")
+	}
```
Comment (samsondav) [inline core/services/validators.go:436]: Maybe consider pulling this block out into another function?

File: core/services/validators.go [hunk 2]
```diff
@@ -392,36 +408,119 @@ func ValidatedOracleSpec(tomlString string) (spec offchainreporting.OracleSpec,
...
-	`schemaVersion`:                          struct{}{},
-	`contractAddress`:                        struct{}{},
-	`p2pPeerID`:                              struct{}{},
-	`p2pBootstrapPeers`:                      struct{}{},
-	`isBootstrapPeer`:                        struct{}{},
-	`blockchainTimeout`:                      struct{}{},
-	`contractConfigTrackerSubscribeInterval`: struct{}{},
-	`contractConfigTrackerPollInterval`:      struct{}{},
-	`contractConfigConfirmations`:            struct{}{},
+// Parameters that must be explicitly set by the operator.
```
Comment (samsondav) [inline core/services/validators.go:440]: For what

File: core/services/validators.go [hunk 3]
```diff
@@ -392,36 +408,119 @@ func ValidatedOracleSpec(tomlString string) (spec offchainreporting.OracleSpec,
...
+			err = multierr.Append(err, errors.Errorf("unrecognised key for %s peer: %s", peerType, k))
 		}
+		delete(expected, k)
 	}
-	return spec, err
+	for missing := range expected {
+		err = multierr.Append(err, errors.Errorf("missing required key %s", missing))
+	}
+	return err
+}
```
Comment (samsondav) [inline core/services/validators.go:488]: Maybe move this function to the bottom?

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/services/validators.go (hunk 2 above)
Comment (connorwstein) [reply core/services/validators.go:440]: Not sure what you mean - just required parameters needed to specify an OCR job

**Review by samsondav — APPROVED** [review]

**Pull request merged**